sstables/sstable_set.cc

```diff
@@ -218,7 +216,8 @@ sstable_set::incremental_selector::select(const dht::ring_position_view& pos) co
 
 sstable_set::incremental_selector
 sstable_set::make_incremental_selector() const {
-    return incremental_selector(_impl->make_incremental_selector(), *_schema);
+    auto selector = _impl->make_incremental_selector();
+    return incremental_selector(std::get<0>(std::move(selector)), std::get<1>(selector));
```

Comment:
auto&& [selector, schema] = _impl->make_incremental_selector();

then you fix the use-after-free above. clang is left to right.

Reply:
actually I take it back. what you wrote should work. the bug is somewhere else.